### ISSUE TITLE
Fixing the pretrain script for Loss Averaging and no_backward_sync() 

### ISSUE THREAD

**Issue by LamOne1** [Lightning-AI/lit-llama#372]
I think #357 should be applied to the pretrain script as well.
https://github.com/Lightning-AI/lit-llama/pull/357
Thank you so much lightning team for this amazing repository.

**Comment by awaelchli** [comment]
The red pajama pretrain script has gradient accumulation already. Shakespeare is missing it, it could be added too there yes. Contributions welcome!

**Comment by LamOne1** [comment]
I'm not yet familiar with GitHub and the code editing, but I'm eager to learn and help out!
I'll try to learn and edit Shakespeare code
